frontend/public/components/overview/index.tsx

```diff
@@ -310,6 +322,27 @@ class OverviewMainContent_ extends React.Component<
     return [...labelSet].sort();
   }
 
+  fetchMonitoringAlerts(): void {
+    const { namespace } = this.props;
+    if (!PROMETHEUS_TENANCY_BASE_PATH) {
+      return;
+    }
+    const poller = (): void => {
+      coFetchJSON(`${PROMETHEUS_TENANCY_BASE_PATH}/api/v1/rules?namespace=${namespace}`)
```

Comment:
You can use ```useURLPoll``` hook here to fetch and watch on API.

Reply:
fetchMonitoringAlerts is being defined inside a class component